### PR TITLE
Forwarding xAPI events to parent window on embed

### PR DESCRIPTION
**Pull request by fnoks**
PR makes it possible to receive xAPI events also when H5P is embedded

### CODE REVIEW

**Comment by thomasmars** [comment]
Looks good, I have created an issue for reviewing you pull request, you can follow it on the issue tracker: https://h5ptechnology.atlassian.net/browse/HFP-287 😘

**Comment by icc** [comment]
I'm a bit concerned about privacy here; if I embed an H5P on my page and then lure folk in by sharing the URL on h5p.org I will automatically be able to gather all their emails.

Should we've perhaps hash the email(like gravatar.com) or something before sending it so that it can be used to uniquely identify each user and to match against existing user databases?

**Comment by falcon-git** [comment]
That is a good point. No data stored on the provider site should be made available through the messages.

**Comment by falcon-git** [comment]
Status: Waiting for a fix to the above consern before this is merged in.

File: js/xapi-parent-forwarder.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// Forward xAPI events to parent window, making it possible to use the
+// following code from parent window to get events:
+// window.addEventListener('message', function (event) { <...> });
+H5P.externalDispatcher.on('xAPI', function (event) {
+  if (event.data && event.data.statement && event.data.statement.actor && event.data.statement.actor.mbox) {
+    if (H5PIntegration.user && H5PIntegration.user.mail_sha1sum) {
+      event.data.statement.actor = {
+        mbox_sha1sum: H5PIntegration.user.mail_sha1sum,
```
Comment (otacke) [inline js/xapi-parent-forwarder.js:8]: @fnoks Greatly appreciated that the actor will be deleted here! Where's sha1 used though? I think it's pretty much broken and should not be used anymore.

Comment (fnoks) [reply js/xapi-parent-forwarder.js:8]: Hmm. mbox_sha1sum is part of the specification, and I didn't see any other options (https://xapi.com/blog/deep-dive-actor-agent/). Maybe we should just add a mail_sha2sum instead? For this (non-LRS) purpose, it probably doesn't matter much. What do you think?

Comment (otacke) [reply js/xapi-parent-forwarder.js:8]: @fnoks Hmm, guess following what is there then is the best that you can do for now given that there's no extension option for the `actor` property.

**Comment by fnoks** [comment]
I am closing my own PR because of privacy concerns. Even if the _actor_ is removed, you never know what the rest of the data might contain, and this becomes especially problematic since "public" H5Ps can be embedded by anyone on other sites (and then might collect data).

**Comment by otacke** [comment]
@fnoks It requires guts to close one's own pull-request, kudos! I assume that if one really wanted to enable this feature, one should require the receiving side to authenticate in some way or another and even then encrypt the data. Kind of LTI light :-)

**Comment by icc** [comment]
Wouldn't just adding a `targetOrigin` to `postMessage()` be sufficient? I.e. whitlisting who can get the data?

**Comment by otacke** [comment]
@icc That would be one form of authentication, yes. AFAIR, you're limited to one domain then though.

**Pull request closed without merging**